system/jhtree/keybuild.cpp

```diff
@@ -482,7 +482,14 @@ class CKeyBuilder : public CInterfaceOf<IKeyBuilder>
                 if (bloomBuilder.valid())
                 {
                     Owned<const BloomFilter> filter = bloomBuilder.build();
-                    writeBloomFilter(*filter, rowHashers.item(idx).queryFields());
+                    writeBloomFilter(*filter);
+                }
+            }
+            if (optBloomFilters)
```

Comment:
clarification: Does bloom filter need to be built (line 484-485) when the optBloomFilters is provided?

Reply:
It is theoretically possible/useful for both to be supplied, so I would keep it.